api/v3/Membership.php

```diff
@@ -42,7 +59,17 @@
  *   API result array.
  */
 function civicrm_api3_membership_delete($params) {
-  return _civicrm_api3_basic_delete(_civicrm_api3_get_BAO(__FUNCTION__), $params);
+  if (isset($params['preserve_contribution'])) {
+    if (CRM_Member_BAO_Membership::del($params['id'], $params['preserve_contribution'])) {
+      return civicrm_api3_create_success(TRUE, $params);
+    }
+    else {
+      return civicrm_api3_create_error('Could not delete membership');
```

Comment:
Hi @joseltorres Hi Jose, I think the better thing to do here would be to throw a exception but not sure anythoughts @eileenmcnaughton